trigger/host_spawn.go

```diff
@@ -272,3 +275,120 @@ func (t *spawnHostStateChangeTriggers) emailPayload(action, result, hostID, url,
 		PlainTextContents: false,
 	}
 }
+
+type spawnHostSetupScriptTriggers struct {
+	hostBase
+}
+
+func makeSpawnHostSetupScriptTriggers() eventHandler {
+	t := &spawnHostSetupScriptTriggers{}
+	t.triggers = map[string]trigger{
+		event.TriggerOutcome: t.spawnHostSetupScriptOutcome,
+	}
+	return t
+}
+
+func (t *spawnHostSetupScriptTriggers) spawnHostSetupScriptOutcome(sub *event.Subscription) (*notification.Notification, error) {
+	if !t.host.UserHost {
+		return nil, nil
+	}
+	payload, err := t.makePayload(sub)
+	if err != nil {
+		return nil, errors.Wrap(err, "making payload")
+	}
+	return notification.New(t.event.ID, sub.Trigger, &sub.Subscriber, payload)
+}
+
+func (t *spawnHostSetupScriptTriggers) makePayload(sub *event.Subscription) (interface{}, error) {
+	if t.event.EventType != event.EventHostScriptExecuteFailed && t.event.EventType != event.EventHostScriptExecuted {
+		return nil, errors.Errorf("unexpected event type '%s'", t.event.EventType)
+	}
+
+	var result string
+	if t.event.EventType == event.EventHostScriptExecuted {
+		result = "succeeded"
+	} else {
+		result = "failed"
+		if strings.Contains(t.data.Logs, evergreen.FetchingTaskDataUnfinishedError) {
+			result = "failed to start"
+		}
+	}
+	var spawnHostScriptPath string
+	if t.host.ProvisionOptions != nil && t.host.ProvisionOptions.TaskId != "" {
+		pRef, err := model.GetProjectRefForTask(t.host.ProvisionOptions.TaskId)
+		if err != nil {
+			return "", errors.Wrap(err, "getting project")
+		}
+		if pRef != nil {
+			spawnHostScriptPath = pRef.SpawnHostScriptPath
+		}
+	}
+
+	switch sub.Subscriber.Type {
+	case event.SlackSubscriberType:
+		return t.slackPayload(spawnHostScriptPath, result, t.host.Id, spawnHostURL(t.uiConfig.Url), hostURL(t.uiConfig.Url, t.host.Id)), nil
+	case event.EmailSubscriberType:
+		return t.emailPayload(spawnHostScriptPath, result, t.host.Id, spawnHostURL(t.uiConfig.Url), hostURL(t.uiConfig.Url, t.host.Id)), nil
+	default:
+		return nil, errors.Errorf("unexpected subscriber type '%s'", sub.Subscriber.Type)
+	}
+}
+
+func (t *spawnHostSetupScriptTriggers) slackPayload(spawnHostScriptPath, result, hostID, url, hostURL string) *notification.SlackPayload {
+	color := evergreenSuccessColor
+	if !t.data.Successful {
+		color = evergreenFailColor
+	}
+
+	attachment := message.SlackAttachment{
+		Title:     "Spawn hosts page",
+		TitleLink: url,
+		Color:     color,
+		Fields: []*message.SlackAttachmentField{
+			{
+				Title: "SSH Command",
+				Value: fmt.Sprintf("`%s`", sshCommand(t.host)),
+			},
+		},
+	}
+	if spawnHostScriptPath != "" {
+		attachment.Fields = append(attachment.Fields,
+			&message.SlackAttachmentField{
+				Title: "With data from task",
+				Value: fmt.Sprintf("<%s|%s>", taskLink(t.uiConfig.Url, t.host.ProvisionOptions.TaskId, -1), t.host.ProvisionOptions.TaskId),
+				Short: true,
+			})
+		attachment.Fields = append(attachment.Fields,
+			&message.SlackAttachmentField{
+				Title: "Setup Script Command",
```

Comment:
nit: `Setup Script Path`? Since it's not the entire script/list of commands, it's just the path to the script on the spawn host.